pkg/lib/serverTime.js

```diff
@@ -321,13 +324,181 @@ export function ServerTime() {
+    function set_ntp_config_timesyncd(config) {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
+
+        const text = `# This file is automatically generated by Cockpit\n\n[Time]\n${config.enabled ? "" : "#"}NTP=${config.servers.join(" ")}\n`;
 
         return cockpit.spawn(["mkdir", "-p", "/etc/systemd/timesyncd.conf.d"], { superuser: "try" })
                 .then(() => custom_ntp_config_file.replace(text));
+    }
+
+    function find_file_and_read(names, first) {
+        if (first === undefined)
+            first = 0;
+        if (first >= names.length)
+            return Promise.reject(new Error("Found none of " + names.join(",")));
+        return cockpit.file(names[first], { superuser: "try" }).read()
+                .then(text => {
+                    if (text)
+                        return [names[first], text];
+                    return find_file_and_read(names, first + 1);
+                });
+    }
+
+    function get_ntp_config_chronyd() {
+        return find_file_and_read(["/etc/chrony.conf", "/etc/chrony/chrony.conf"])
+                .then(([name, text]) => {
+                    const result = {
+                        supported: true,
+                        backend: "chronyd",
+                        file: name,
+                        standard: [],
+                        sources: [],
+                        extra: []
+                    };
+
+                    let in_cockpit_section = false;
+                    for (const line of text.split("\n")) {
+                        if (line == "# Added by Cockpit") {
+                            in_cockpit_section = true;
+                        } else {
+                            const fields = line.split(" ");
+                            if (fields[0] == "server") {
+                                if (in_cockpit_section) {
+                                    result.extra.push(fields[1]);
+                                } else {
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                }
+                            } else {
+                                in_cockpit_section = false;
+                                if (fields[0] == "pool")
+                                    result.standard.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#pool" || fields[0] == "#server")
+                                    result.standard.push({ name: fields[1], enabled: false });
+                                if (fields[0] == "sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: true });
+                                if (fields[0] == "#sourcedir")
+                                    result.sources.push({ name: fields[1], enabled: false });
```

Comment:
This function also needs to read all the `confdir` files.

Reply:
Yes...